core/src/main/java/com/scalar/db/storage/jdbc/query/SimpleSelectQuery.java

```diff
@@ -86,7 +86,9 @@ private String conditionSqlString() {
   }
 
   private String orderBySqlString() {
-    if (!isRangeQuery || indexedColumn.isPresent()) {
+    if (!isRangeQuery
+        || indexedColumn.isPresent()
+        || tableMetadata.getClusteringKeyNames().isEmpty()) {
```

Comment:
We always added ORDER BY clause when we scan a table (except index scan) but if the table doesn't have clustering key, we should not have done that.